File: src/main/resources/static/app/classes/Message.ts
/**
 * Class that represent the information related to MiausicBox messages
 * @class Message
 */
import { User } from './User';

export class Message {
  
  /* Attributes */
  private _id: number;
  private _sender: User;
  private _destiny: User;
  private _subject: string;
  private _message: string;
  private _date: Date;
  private _readd: boolean;

  /* Constructor */
  constructor(id:number, sender:User, destiny:User, subject:string, message:string, date:Date, readd:boolean) {
    this._id = id;
    this._sender = sender;
    this._destiny = destiny;
    this._subject = subject;
    this._message = message;
    this._date = date;
    this._readd = readd;
  }

  /* Getters & Setters */
  get id():number {
    return this._id;
  }

  get sender():User {
    return this._sender;
  }

  get destiny():User {
    return this._destiny;
  }

  get subject():string {
    return this._subject;
  }


  get message():string {
    return this._message;
  }

  get date():Date {
    return this._date;
  }

  get readd():boolean {
    return this._readd;
  }

  set id(id:number) {
    this._id = id;
  }

  set sender(sender:User) {
    this._sender = sender;
  }

  set destiny(destiny:User) {
    this._destiny = destiny;
  }

  set subject(value:string){
    this._subject=value;
  }

  set message(message:string) {
    this._message = message;
  }

  set date(date:Date) {
    this._date = date;
  }

  set readd(value:boolean) {
    this._readd=value;
  }

  /** Return if an Message is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any) {
    if (!(object instanceof Message)) {
      return false;
    } else {
      var message:Message = object;
      return (message.sender.equals(this.sender) &&
              message.destiny.equals(this.destiny) &&
              message.message == this.message);
    }
  }

}

File: src/main/java/giraffe/miausicbox/model/Message.java
package giraffe.miausicbox.model;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

import giraffe.miausicbox.user.User;

@Entity
public class Message {

	public interface Basic extends User.Basic {}

	// Attributes
	@JsonView(Basic.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(Basic.class)
	@OneToOne
	private User sender;
	
	@JsonView(Basic.class)
	@OneToOne
	private User destiny;
	
	@JsonView(Basic.class)
	private String subject;
	
	@JsonView(Basic.class)
	private String message;
	
	@JsonView(Basic.class)
	private String date;
	
	@JsonView(Basic.class)
	private Boolean readd;
	
	// Constructor
	public Message() {}
	
	public Message(
			User sender,
			User destinty,
			String subject,
			String message,
			String date,
			Boolean readd
			) {
		super();
		this.sender = sender;
		this.destiny = destinty;
		this.message = message;
		this.date = date;
		this.readd = readd;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public User getSender() {
		return sender;
	}

	public void setSender(User sender) {
		this.sender = sender;
	}

	public User getDestiny() {
		return destiny;
	}

	public void setDestiny(User destiny) {
		this.destiny = destiny;
	}

	public String getSubject() {
		return subject;
	}

	public void setSubject(String subject) {
		this.subject = subject;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public String getDate() {
		return date;
	}

	public void setDate(String date) {
		this.date = date;
	}

	public Boolean getReadd() {
		return readd;
	}

	public void setReadd(Boolean readd) {
		this.readd = readd;
	}

	// ToString
	@Override
	public String toString() {
		return (this.getSender() + " - " + this.getDestiny()
				+ " (" + this.getDate().toString() + ")");
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Message other = (Message) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getSender(), other.getSender())
				&& Objects.equals(this.getDestiny(), other.getDestiny())
				&& Objects.equals(this.getDate(), other.getDate()));
	}

}
